framework/src/org/checkerframework/framework/util/typeinference/DefaultTypeArgumentInference.java

```diff
@@ -113,15 +120,33 @@
         final AnnotatedTypeMirror assignedTo =
                 TypeArgInferenceUtil.assignedTo(typeFactory, typeFactory.getPath(expressionTree));
 
+        if (showInferenceSteps) {
+            System.err.println("DTAI: expression: " + expressionTree.toString().replace("\n", " "));
```

Comment:
Changed it to use the javac messager.
This revealed the change in https://github.com/typetools/checker-framework/pull/1289 , which is included here to make this code compile.